aptos-move/mvhashmap/src/versioned_group_data.rs

```diff
@@ -246,7 +246,11 @@ impl<T: Hash + Clone + Debug + Eq + Serialize, V: TransactionWrite> VersionedGro
     }
 
     // Records the latest committed op for each tag in the group (removed tags ar excluded).
-    fn commit_idx(&mut self, shifted_idx: ShiftedTxnIndex) -> anyhow::Result<()> {
+    fn commit_idx(
+        &mut self,
+        shifted_idx: ShiftedTxnIndex,
+        allow_new_modification: bool,
```

Comment:
What is this flag for?
Why wouldn't allow modification of a tag?

Reply:
We create sentinels as modifications now, without this flag committing zero index gets an invariant violation error.
Having sentinels as modifications is useful when replacing reads with writes, since the kind should be modification. Previously we had Creations and then ugly logic to turn it into modification